pkg/api/v1/atlasdeployment_types.go

```diff
@@ -205,9 +214,19 @@ type ServerlessSpec struct {
 	// +kubebuilder:validation:Pattern:=^[a-zA-Z0-9][a-zA-Z0-9-]*$
 	Name string `json:"name"`
 	// Configuration for the provisioned hosts on which MongoDB runs. The available options are specific to the cloud service provider.
-	ProviderSettings *ProviderSettingsSpec `json:"providerSettings"`
-
+	ProviderSettings *ProviderSettingsSpec       `json:"providerSettings"`
 	PrivateEndpoints []ServerlessPrivateEndpoint `json:"privateEndpoints,omitempty"`
+	// Key-value pairs for resource tagging.
+	// +kubebuilder:validation:MaxItems=50
+	// +optional
+	Tags []*TagSpec `json:"tags"`
```

Comment:
An alternative way to model this with types would have been doing this here:

```suggestion
	Tags TagSpecSet `json:"tags"`
```

Where:
```
type TagSpecSet []*TagSpec

// Validate returns an error is TagSpecSet is not a proper set, keys are repeated.
func (tset *TagSpecSet) Validate() error {
...

// Equal returns true if the given TagSet holds the same key value pairs, regardless or ordering
func (tset *TagSpecSet) Equal(other *TagSpecSet) bool {
...
```

Reply:
While I got the point here, I'd say calling a slice a Set is a bad name choice :smile: